### PR TITLE
implements Zookeeper weighted quorum and improve readinessProbe, to avoid wrong quorum election and lost data in zookeeper

### PR DESCRIPTION
**Pull request by Slach**
Signed-off-by: Slach <bloodjazman@gmail.com>

## Important items to consider before making a Pull Request
Please check items PR complies to:
* [x] All commits in the PR are squashed. [More info](https://github.com/Altinity/clickhouse-operator/blob/master/CONTRIBUTING.md#how-to-make-pr) 
* [x] The PR is made into dedicated `next-release` branch, **not into** `master` branch<sup>1</sup>. [More info](https://github.com/Altinity/clickhouse-operator/blob/master/CONTRIBUTING.md#how-to-make-pr)
* [x] The PR is signed. [More info](https://github.com/Altinity/clickhouse-operator/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: deploy/zookeeper/quick-start-persistent-volume/zookeeper-1-node.yaml [hunk 1]
```diff
@@ -170,14 +170,47 @@ spec:
               for (( i=1; i<=$SERVERS; i++ )); do
                   echo "server.$i=$NAME-$((i-1)).$DOMAIN:$SERVER_PORT:$ELECTION_PORT" >> /conf/zoo.cfg;
               done &&
+              if [[ $SERVERS -eq 1 ]]; then
```
Comment (alex-zaitsev) [inline deploy/zookeeper/quick-start-persistent-volume/zookeeper-1-node.yaml:173]: Do we need it for 1-node setup?

File: deploy/zookeeper/quick-start-persistent-volume/zookeeper-1-node-1GB-for-tests-only.yaml [hunk 2]
```diff
@@ -158,14 +158,47 @@ spec:
               for (( i=1; i<=$SERVERS; i++ )); do
                   echo "server.$i=$NAME-$((i-1)).$DOMAIN:$SERVER_PORT:$ELECTION_PORT" >> /conf/zoo.cfg;
               done &&
+              if [[ $SERVERS -eq 1 ]]; then
```
Comment (alex-zaitsev) [inline deploy/zookeeper/quick-start-persistent-volume/zookeeper-1-node-1GB-for-tests-only.yaml:161]: Do we need it for a single node?

Comment (Slach) [reply deploy/zookeeper/quick-start-persistent-volume/zookeeper-1-node-1GB-for-tests-only.yaml:161]: yes, cause on ACM side we always use `1-node.yaml` and just replace `replicas: X` and `SERVERS=X`

**Comment by alex-zaitsev** [comment]
I suggest not to merge it into clickhouse-operator at all, but move relevant ZK manifests to ACM. Otherwise, we are testing in production.

**Pull request merged**